### ISSUE TITLE
How to enable text selection in PDF

### ISSUE THREAD

**Issue by nicolls1** [wojtekmaj/react-pdf#764]
**Before you start - checklist**

 - [x] I have read documentation in README
 - [x] I have checked [sample](https://github.com/wojtekmaj/react-pdf/tree/master/sample) and [test](https://github.com/wojtekmaj/react-pdf/tree/master/test) suites to see real life basic implementation
 - [x] I have checked if this question is not already asked

**What are you trying to achieve? Please describe.**

How to allow users to select text? I've found issue #640 which seems to be opposite of what I'm trying to do. My end goal is to use the brower's `selectionchange` and `window.getSelection()` to get the text the user has selected.

The text on the sample site (https://projects.wojtekmaj.pl/react-pdf/) is selectable and I am trying to reproduce that functionality. 

**Describe solutions you've tried**

I have the starting code from the readme: 
```
  const [numPages, setNumPages] = useState<number | null>(null)
  const [pageNumber, setPageNumber] = useState(1)

  function onDocumentLoadSuccess({ numPages }: { numPages: number }) {
    setNumPages(numPages)
  }

  return (
    <div>
      <Document
        file="./test2.pdf"
        onLoadSuccess={onDocumentLoadSuccess}
        renderMode="svg"
      >
        <Page pageNumber={pageNumber} renderTextLayer={true} />
      </Document>
      <p>
        Page {pageNumber} of {numPages}
      </p>
    </div>
  )
```

I tried using svg instead of canvas as the render mode.

I found the pdf used on the demo site (https://projects.wojtekmaj.pl/react-pdf/) and tried using that one. It seems that SVG paths are used everywhere so I can't select. 
<img width="897" alt="Screenshot 2021-04-12 at 15 48 09" src="https://user-images.githubusercontent.com/2540582/114404959-7e410980-9ba6-11eb-81f4-0e4198df3783.png">

When I look at the demo site, it has html elements. How is this being done?

**Environment**

 - Browser (if applicable) [e.g. Chrome 57, Firefox 59]: Chrome 89
 - React-PDF version [e.g. 3.0.4]: 5.2.0
 - React version [e.g. 16.3.0]: 17.0.1

**Comment by nicolls1** [comment]
After looking into it a bit more, I have found out this is related to #556. I was not aware of the invisible text layer on top of the canvas but when I found it, it was clear that it did not align with the content under it. For me, the following alligned the text layer: 
```
.react-pdf__Page__svg {
  margin: 0 auto;
}
```

**Comment by CodeAnnihilator** [comment]
When I installed react-pdf version 6.2.2, I also had to copy file `node_modules/pdfjs-dist/build/pdf.worker.js` into `public/pdf.worker.js` and add property `options={{ workerSrc: 'pdf.worker.js' }}` to the Document component as per this article [https://levelup.gitconnected.com/displaying-pdf-in-react-app-6e9d1fffa1a9](url), but then I faced issue that I couldn't select any text.

I realised that whole text was shifted under rendered PDF file. I was wondering how it works here [https://codesandbox.io/s/displaying-pdf-using-react-5d003](url), but does not work in my code, even though everything looks the same.

So I decided to check version of react-pdf at this link [https://codesandbox.io/s/displaying-pdf-using-react-5d003](url) and found out that version in package.json is 4.1.0, so I installed the same version in my application, and also (!) copied pdf.worker.js file into my code from this codesandbox example above (because this file has different code, based on react-pdf version), and it worked.

Now I can select text, and it is properly aligned.
